lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSensitivity.cxx

```diff
@@ -47,38 +48,160 @@ FunctionalChaosRandomVector::FunctionalChaosRandomVector(const FunctionalChaosRe
+  for (UnsignedInteger m = 0; m < outputDimension; ++ m)
+  {
+    UnsignedInteger maxdegree = 0;
+
+    // compute part of contribution of each basis term
+    for (UnsignedInteger i = 0; i < basisSize; ++ i)
+    {
+      Scalar coefI = coefficients[i][m];
+      Indices multiIndices(enumerateFunction(indices[i]));
+      UnsignedInteger degreeI = 0;
+      for (UnsignedInteger k = 0; k < multiIndices.getSize(); ++ k) degreeI += multiIndices[k];
+
+      maxdegree = std::max(maxdegree, degreeI);
+      if (i > 0)
+      {
+        varianceRatio[i] = coefI * coefI / covariance(m, m);
+      }
+    }
+
+    // sort basis terms by descending variance contribution
+    Indices varianceOrder(basisSize);
+    varianceOrder.fill();
+    for (UnsignedInteger i = 1; i < basisSize; ++ i)
+    {
+      for (UnsignedInteger j = basisSize - 1; j >= i; -- j)
+      {
+        if (varianceRatio[varianceOrder[j]] < varianceRatio[varianceOrder[j + 1]])
+        {
+          UnsignedInteger tmp = varianceOrder[j];
+          varianceOrder[j] = varianceOrder[j + 1];
+          varianceOrder[j + 1] = tmp;
+        }
+      }
+
+      // stop when the variance contribution becomes less than epsilon
+      if (varianceRatio[varianceOrder[i]] < ResourceMap::GetAsScalar("FunctionalChaosSensitivity-VariancePartThreshold"))
+      {
+        varianceOrder.resize(i);
+        break;
+      }
+    }
+
+    if (outputDimension > 1)
+      oss << "Marginal: " << m << "\n";
+
+    // table of part of variance for each basis term
+    oss << "Index   | Multi-indice                  | Part of variance  \n";
+    oss << String(60, '-') << "\n";
+    for (UnsignedInteger i = 0; i < varianceOrder.getSize(); ++i)
+    {
+      Indices multiIndices(enumerateFunction(indices[varianceOrder[i]]));
+      UnsignedInteger degreeI = 0;
+      for (UnsignedInteger k = 0; k < multiIndices.getSize(); ++ k) degreeI += multiIndices[k];
+
+      maxdegree = std::max(maxdegree, degreeI);
+      st = OSS() << std::setw(7) << varianceOrder[i];
+      oss << st << " | ";
+
+      st = OSS() << multiIndices;
+      oss << st << String(29-st.size(), ' ') << " | ";
+
+      if (i > 0)
+      {
+        st = OSS() << varianceRatio[varianceOrder[i]];
+      }
+      else
+        st = "N/A";
+      oss << st << "\n";
+    }
+    oss << String(60, '-') << "\n";
+    oss << "\n\n";
+
+    // table of first/total order indices for each input
+    oss << String(60, '-') << "\n";
+    oss << "Component | Sobol index            | Sobol total index      \n";
+    oss << String(60, '-') << "\n";
+    for (UnsignedInteger i = 0; i < inputDimension; ++ i)
+    {
+      st = OSS() << std::setw(9) << i;
+      oss << st << " | ";
+
+      st = OSS() << getSobolIndex(i, m);
+      oss << st << String(22-st.size(), ' ') << " | ";
+
+      st = OSS() << getSobolTotalIndex(i, m);
+      oss << st << String(22-st.size(), ' ') << "\n";
+    }
+    oss << String(60, '-') << "\n";
+    oss << "\n";
+
+  }
+  return oss;
```

Comment:
Definitely NO! Too much complex, and too much inefficient. It should rely on getSobolIndex(), should you improve its interface for multiple indices computation (and even introducing parallelism).